evm-contracts/src/v0.6/dev/DeviationFlaggingValidator.sol

```diff
@@ -162,5 +155,58 @@ contract DeviationFlaggingValidator is Owned, AggregatorValidatorInterface {
     return uint256(value < 0 ? value.mul(-1): value);
   }
 
+  /**
+   * @dev Subtracts two signed integers, returns false 2nd param on overflow.
+   * Modified version of OpenZeppelin's SignedSafeMath.
+   */
+  function safeSub(int256 a, int256 b) internal pure returns (int256, bool) {
+    int256 c = a - b;
+    if ((b < 0 && c <= a) || (b >= 0 && c > a)) {
+      return (0, false);
+    }
+
+    return (c, true);
+  }
+
+
+  /**
+   * @dev Multiplies two signed integers, returns false 2nd param on overflow.
+   * Modified version of OpenZeppelin's SignedSafeMath.
+   */
+  function safeMul(int256 a, int256 b) internal pure returns (int256, bool) {
+    // Gas optimization: this is cheaper than requiring 'a' not being zero, but the
+    // benefit is lost if 'b' is also tested.
+    // See: https://github.com/OpenZeppelin/openzeppelin-contracts/pull/522
+    if (a == 0) {
+      return (0, true);
+    }
+
+    if (a == -1 && b == INT256_MIN) {
+      return (0, false);
+    }
+
+    int256 c = a * b;
+    if (!(c / a == b)) {
+      return (0, false);
+    }
+
+    return (c, true);
+  }
+
+  /**
+   * @dev Divides two signed integers, returns false 2nd param on overflow.
+   * Modified version of OpenZeppelin's SignedSafeMath.
+   */
+  function safeDiv(int256 a, int256 b) internal pure returns (int256, bool) {
+    // don't need to check for 0 as that is checked in isValid
```

Comment:
We should bring back the check for 0, and extract this as a library so it can be used standalone in some other context, when required.